app/src/main/java/com/example/sopt_seminar/util/BaseFragment.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.sopt_seminar.util
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.fragment.app.Fragment
+
+abstract class BaseFragment<T : ViewDataBinding>(
+    @LayoutRes private val layoutId: Int
+) : Fragment() {
+    private var _binding: T? = null
+    val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding =
+            DataBindingUtil.inflate(layoutInflater, layoutId, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.lifecycleOwner = this
```

Comment:
프래그먼트의 lifecycleOnwer를 this로 지정하면 `onCreate-onDestroy` 범위 까지 라이프사이클 오너가 지정되므로

```suggestion
        binding.lifecycleOwner = viewLifeCycleOwner
```

로 지정하는게 좋습니다. 프래그먼트 생명주기 문제로 이것 또한 메모리 릭이 발생할 수 있으니 참고해주세요!

링크1 : https://pluu.github.io/blog/android/2020/01/25/android-fragment-lifecycle/

Reply:
정독했습니다!! 감사합니다!!!